### DISCUSSION TITLE
Server Rendered app + HMR Asset Workflow?

### DISCUSSION THREAD

**Discussion by wesbos** [parcel-bundler/parcel#7283] [Q&A]
I'm working on an app that is traditionally server rendered, it runs on port :3333

Then I run parcel with multiple `targets` in my package.json to bundle a few different CSS files and a few different JS files.

It all works, but I can't get HMR to work and have to refresh to see any JS or CSS changes.

The console says `Firefox can’t establish a connection to the server at ws://courses.wes:3333/.` which makes sense, because the dev server is running on port `:1234`, not `:3333`.

I've fiddled around with the --hmr-port flag but can't figure it out. It looks like the JS loaded on my page is looking for the websocket, but on the wrong port. How do I tell it to listen on `:1234`?

**Comment by mischnic** [comment]
Are you using `parcel watch` or `parcel serve`? Are you doing `<script src="parcel_dist/index.js">` or `<script src="localhost:1234/index.js">`?

**Reply by wesbos** [reply]
I'm using `parcel` to start it, which I assume it parcel serve? 

I'm using `<script src="parcel_dist/index.js">`, should I link it from the dev server instead?

**Reply by wesbos** [reply]
I tried linking it directly to `http://localhost:1234/javascripts/app.js` and it does load the JS, but the Websocket error persists

**Comment by wesbos** [comment]
So, an update, I seem to be able to get it to work if I specify `--hmr-port 7777 --host localhost` in my build command. 

`7777` isn't a port I'm even using, but it works? If i use `3333` (my server port) or `1234` (parcel dev server), it doesn't work.

**Reply by mischnic** [reply]
That is probably a consequence of https://github.com/parcel-bundler/parcel/pull/6005.

>  If i use 3333 (my server port) or 1234 (parcel dev server), it doesn't work.

3333 would override the socket in the other process, so that doesn't work.
And setting it to 1234 triggers the condition in the linked PR.

**Reply by wesbos** [reply]
Thanks, yes it looks like the issue is logged here: https://github.com/parcel-bundler/parcel/issues/6061

**Comment by dpavlic** [comment]
I've been banging my head to have HMR work with Webpack + Wordpress + React for days now (it works fine with Vue, but that's a different unrelated story). I've tried running Parcel and I ran into the same issue, and the (very counter-intuitive) solution of setting a "nonsense" port works! 

EDIT: Sorry to hijack, but I guess I realized now what's happening. Setting it to a random 'other' port sets HMR only to an entirely different port (e.g., 7777), which results in parcel to point at it AND as a consequence it now also works 'embedded' within Wordpress. It does work and that makes me very happy.

**Reply by wesbos** [reply]
So if I understand Parcel is running  a websocket server on the third port? 7777 in my case

**Reply by dpavlic** [reply]
Yep, that's what it looks like.

**Comment by e-krebs** [comment]
the `--hmr-port` option indeed works, but it isn't documented at all on the site.
is there a process to require documentation? (or to actually write it myself)
